handler/randomizer.go

```diff
@@ -0,0 +1,61 @@
+package handler
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	k8upv1alpha1 "github.com/vshn/k8up/api/v1alpha1"
+	"math/big"
+)
+
+const (
+	ScheduleHourlyRandom   = "@hourly-random"
+	ScheduleDailyRandom    = "@daily-random"
+	ScheduleYearlyRandom   = "@yearly-random"
+	ScheduleAnnuallyRandom = "@annually-random"
+	ScheduleMonthlyRandom  = "@monthly-random"
+	ScheduleWeeklyRandom   = "@weekly-random"
+)
+
+func (s *ScheduleHandler) createSeed(schedule *k8upv1alpha1.Schedule, jobType k8upv1alpha1.JobType) string {
+	return schedule.Namespace + "/" + schedule.Name + "@" + string(jobType)
+}
+
+// randomizeSchedule randomizes the given originalSchedule with a seed. The originalSchedule has to be one of the supported
+// '@x-random' predefined schedules, otherwise it returns an error with the original schedule unmodified.
+func randomizeSchedule(seed, originalSchedule string) (string, error) {
+	checksum := calculateChecksumFromSeed(seed)
+
+	minute := new(big.Int).Set(checksum).Mod(checksum, big.NewInt(60))
+	hour := new(big.Int).Set(checksum).Mod(checksum, big.NewInt(24))
```

Comment:
Do not calculate values not needed later on. Depending on the branch the switch statement takes, this and the following variables are not needed in all cases.

Reply:
minute and hour is used in all cases, and dayOfMonth in 2 out of 5. Considering there's a explanation comment I am fine with having dayOfMonth also calculated in all cases in order to avoid duplication.